### ISSUE TITLE
Enhancement request: Allow skipping preprocessor lines.

### ISSUE THREAD

**Issue by kbauer** [DarthFennec/highlight-indent-guides#104]
In some languages, preprocessor directives are used, most commonly C-style. These directives always start at the beginning of the line. This can lead to code like:

```
int doSomething() {
  if(1) {
    if(2) {
      if(3) {
        hello();
        world();
#ifdef DEBUG
        printInfo(1);
        printInfo(2);
#endif
        hello();
        world();
      }
    }
  }
}
```

## Expectation

Consistent guides across the directives.

![image](https://user-images.githubusercontent.com/7498468/99124032-a6c9d500-2601-11eb-8c2f-0b912ce372b5.png)
(Mockup)

## Result

The preprocessor lines confuse the guide creation, since they interrupt the indentation.

![image](https://user-images.githubusercontent.com/7498468/99124173-f27c7e80-2601-11eb-8c05-141556ceecea.png)

## Possible solutions?

It likely is a problem without a good general solution; Interruptions of a strict "logical level = indent level" pattern may occur differently from code base to code base, even within the same language.

The best bet would likely be to allow configuring a regexp for lines to be ignored, which could then be customized per-mode, per-file or per-directory through mode-hooks, directory-local variables and file-local variables.

**Comment by DarthFennec** [comment]
I agree, there will need to be some kind of customizable regex or function for this. I'll look into it when I have time, thanks!
